### ISSUE TITLE
aria-labelledby on a div without a role is not useful

### ISSUE THREAD

**Issue by jcoyne** [projectblacklight/blacklight#3119]
I think we should revert https://github.com/projectblacklight/blacklight/commit/b787f3710f10ef1a7a2806653cbf3409dbb6d504

Using aria-labelledby in this way is no longer recommended by Bootstrap https://getbootstrap.com/docs/5.3/components/accordion/ (removed in https://github.com/twbs/bootstrap/pull/37766)

Siteimprove is flagging this for us:
<img width="332" alt="Screenshot 2024-01-17 at 2 31 32 PM" src="https://github.com/projectblacklight/blacklight/assets/92044/136ddfd4-f34f-4a15-a15d-aa817adcfeed">

> aria-label does not work on `<div>` nor `<span>`.
>
> A `<div>` and `<span>` are neither landmarks nor interactive content. An aria-label will not be read by a screen reader (and rightly so).

https://stackoverflow.com/a/42682268/162852

Another possibility is adding `role="group"`

**Comment by dbranchini** [comment]
I need to dig a bit more on this, because I use this site a lot for aria guidance - https://www.w3.org/WAI/WCAG21/Techniques/#aria, and I found this (with an alert that their example still needs additional testing) - https://www.w3.org/WAI/ARIA/apg/patterns/accordion/examples/accordion/ where they are using aria-labelledby in the exact same way as we are. Yet, I see how SiteImprove is picking this up as an error too.

**Comment by jcoyne** [comment]
@dbranchini it looks like the accordion example there uses `role="region"`, which seems like a reasonable solution if you agree.

**Comment by dbranchini** [comment]
Yes, that sounds like a good solution.
